x/evm/keeper/msg_server.go

```diff
@@ -633,21 +637,26 @@ func (s msgServer) ConfirmTransferKey(c context.Context, req *types.ConfirmTrans
 func (s msgServer) VoteConfirmChain(c context.Context, req *types.VoteConfirmChainRequest) (*types.VoteConfirmChainResponse, error) {
 	ctx := sdk.UnwrapSDKContext(c)
 
-	registeredChain, registered := s.nexus.GetChain(ctx, req.Name)
-	if registered {
-		return &types.VoteConfirmChainResponse{Log: fmt.Sprintf("chain %s already confirmed", registeredChain.Name)}, nil
-	}
-	chain, ok := s.GetPendingChain(ctx, req.Name)
-	if !ok {
-		return nil, fmt.Errorf("unknown chain %s", req.Name)
-	}
-
 	voter := s.snapshotter.GetOperator(ctx, req.Sender)
 	if voter == nil {
 		return nil, fmt.Errorf("account %v is not registered as a validator proxy", req.Sender.String())
 	}
 
+	chain, chainFound := s.GetPendingChain(ctx, req.Name)
+
 	poll := s.voter.GetPoll(ctx, req.PollKey)
+	switch {
+	case poll.Is(vote.Expired):
+		return &types.VoteConfirmChainResponse{Log: fmt.Sprintf("vote for poll %s already expired", req.PollKey)}, nil
+	case poll.Is(vote.Failed), poll.Is(vote.Completed):
+		// if the voting threshold has been met and additional votes are received they should not return an error
+		return &types.VoteConfirmChainResponse{Log: fmt.Sprintf("vote for poll %s already decided", req.PollKey)}, nil
+	default:
+		if !chainFound {
+			return nil, fmt.Errorf("unknown chain %s", req.Name)
+		}
```

Comment:
do this after the switch, then you can also move GetPendingChain to after the switch